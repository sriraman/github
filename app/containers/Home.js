import React, { Component } from 'react'
import { View, Text, TextInput, Image, TouchableOpacity } from 'react-native'
import baseStyles from '../styles/base'
import { Actions } from 'react-native-router-flux'

class Home extends Component {
	constructor (props) {
		super(props)
		this.state = {
			repo: 'facebook/react-native',
			username: '',
			password: '',
			errorMessage: null
		}
	}
	_go () {
		this.setState({
			errorMessage: null
		})
		Actions.commitList({
			username: this.state.username,
			password: this.state.password,
			repo: this.state.repo
		})
	}
	componentWillReceiveProps (nextProps) {
		if (nextProps.errorMessage) {
			this.setState({
				errorMessage: nextProps.errorMessage
			})
		}
	}
	render () {
		return <View style={baseStyles.container}>
			<Image source={{ uri: 'https://assets-cdn.github.com/images/modules/logos_page/Octocat.png' }} style={baseStyles.logo} />
			{this.state.errorMessage &&
				<Text style={baseStyles.errorMessage}> {this.state.errorMessage} </Text>
			}
			<View style={baseStyles.textboxContainer}>
				<TextInput onChangeText={(username) => this.setState({ username })} style={baseStyles.textbox} value={this.state.username} placeholder="Username" underlineColorAndroid="transparent"/>
			</View>
			<View style={baseStyles.textboxContainer}>
				<TextInput onChangeText={(password) => this.setState({ password })} style={baseStyles.textbox} value={this.state.password} placeholder="Password" underlineColorAndroid="transparent" secureTextEntry={true}/>
			</View>
			<View style={baseStyles.textboxContainer}>
				<TextInput onChangeText={(repo) => this.setState({ repo })} style={baseStyles.textbox} value={this.state.repo} placeholder="Repo Name" underlineColorAndroid="transparent"/>
			</View>
			<TouchableOpacity onPress={this._go.bind(this)} style={baseStyles.goButton}>
					<Text style={baseStyles.goButtonText}> Go </Text>
			</TouchableOpacity>
		</View>
	}
}

module.exports = Home
