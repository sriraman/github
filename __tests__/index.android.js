import 'react-native'
import React from 'react'
import Home from '../app/containers/Home'

import renderer from 'react-test-renderer'

/**
 * To Mock the native module
 */
jest.mock('Linking', () => {
	return {
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
		openURL: jest.fn(),
		canOpenURL: jest.fn(),
		getInitialURL: jest.fn()
	}
})

/**
 * Home Page
 */
it('renders correctly', () => {
	const tree = renderer.create(
		<Home />
	).toJSON()
	expect(tree).toMatchSnapshot()
})
