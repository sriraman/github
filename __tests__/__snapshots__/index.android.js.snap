// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://assets-cdn.github.com/images/modules/logos_page/Octocat.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 150,
        "width": 150,
      }
    }
  />
  <View
    style={
      Object {
        "borderColor": "#000",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "margin": 10,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Username"
      style={
        Object {
          "flex": 1,
          "paddingLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "borderColor": "#000",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "margin": 10,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "flex": 1,
          "paddingLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "borderColor": "#000",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "margin": 10,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Repo Name"
      style={
        Object {
          "flex": 1,
          "paddingLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
      value="facebook/react-native"
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000",
        "borderRadius": 10,
        "margin": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 22,
          "textAlign": "center",
        }
      }
    >
       Go 
    </Text>
  </View>
</View>
`;
